epochStart/metachain/economics.go

```diff
@@ -185,6 +185,13 @@ func (e *economics) ComputeEndOfEpochEconomics(
 		rewardsForProtocolSustainability,
 	)
 
+	err = e.checkEconomicsInvariants(computedEconomics, inflationRate, maxBlocksInEpoch, totalNumBlocksInEpoch, metaBlock, metaBlock.Epoch)
+	if err != nil {
+		log.Warn("ComputeEndOfEpochEconomics", "error", err.Error())
+		// TODO: return maybe old epoch economics instead of failing?
```

Comment:
Open for debate: we can return the old economics or a computed economics struct holding 0 rewards/accumulated fees and such.

Reply:
Let's discuss and implement in a follow up PR.